frontend/src/views/main/project/TextExplanation.vue

```diff
@@ -50,80 +50,72 @@
   </div>
 </template>
 
-<script lang="ts">
-import {Component, Prop, Vue, Watch} from 'vue-property-decorator';
-import OverlayLoader from '@/components/OverlayLoader.vue';
-import {api} from '@/api';
+<script lang="ts" setup>
+import {ref, watch} from "vue";
 import mixpanel from "mixpanel-browser";
+import {api} from "@/api";
+import OverlayLoader from "@/components/OverlayLoader.vue";
 
-@Component({
-  components: {OverlayLoader},
-})
-export default class TextExplanation extends Vue {
-  @Prop({required: true}) modelId!: number;
-  @Prop({required: true}) datasetId!: number;
-  @Prop({required: true}) textFeatureNames!: string[];
-  @Prop({required: true}) classificationLabels!: string[];
-  @Prop({default: {}}) inputData!: object;
-  @Prop() classificationResult!: string;
+interface Props {
+  modelId: number,
+  datasetId: number,
+  textFeatureNames: string[],
+  classificationLabels: string[]
+  inputData?: object,
+  classificationResult: string
+}
 
-  loading: boolean = false;
-  originalInputData = {}; // used for the watcher to trigger the explanation call
-  selectedFeature = "";
-  selectedLabel = this.classificationResult || this.classificationLabels[0];
-  errorMsg: string = "";
-  result: { [key: string]: string } = {};
-  submitted = false;
+const props = withDefaults(defineProps<Props>(), {
+  inputData: () => {}
```

Comment:
```suggestion
  inputData: () => ({})
```

otherwise the function returns undefined